examples/example.c

```diff
@@ -0,0 +1,248 @@
+    printf("tklock: valid=%s mode=%s locked=%s\n",
+           bool_repr(tklock->valid),
+           tklock_mode_repr(tklock->mode),
+           bool_repr(tklock->locked));
+}
+
+static void inactivity_cb(MceInactivity *inactivity, void *arg)
+{
+    (void)arg;
+    printf("inactivity: valid=%s status=%s\n",
+           bool_repr(inactivity->valid),
+           bool_repr(inactivity->status));
+}
+
+/* ========================================================================= *
+ * MAIN_ENTRY
+ * ========================================================================= */
+
+static gboolean quit_cb(gpointer aptr)
+{
+    printf("quit\n");
+    mainloop_quit();
+    guint *id_ptr = aptr;
+    *id_ptr = 0;
+    return G_SOURCE_REMOVE;
+}
+
+int
+main(int argc, char **argv)
+{
+    printf("startup\n");
+
+    int            exitcode       = EXIT_FAILURE;
+    gint           timeout_s      = 0;
+    guint          timeout_id     = 0;
+    guint          sigterm_id     = 0;
+    guint          sigint_id      = 0;
+    MceBattery    *mce_battery    = mce_battery_new();
+    MceCharger    *mce_charger    = mce_charger_new();
+    MceDisplay    *mce_display    = mce_display_new();
+    MceInactivity *mce_inactivity = mce_inactivity_new();
+    MceTklock     *mce_tklock     = mce_tklock_new();
+
+    /* Note: As handler callbacks do not use user data -> implicit signal
+     *       disconnect at final gobject unref is enough and there is no
+     *       need to keep track of added signal handler ids. */
+    mce_battery_add_valid_changed_handler(mce_battery, battery_cb,  NULL);
+    mce_battery_add_level_changed_handler(mce_battery, battery_cb,  NULL);
+    mce_battery_add_status_changed_handler(mce_battery, battery_cb, NULL);
+
+    mce_charger_add_valid_changed_handler(mce_charger, charger_cb, NULL);
+    mce_charger_add_state_changed_handler(mce_charger, charger_cb, NULL);
+
+    mce_display_add_valid_changed_handler(mce_display, display_cb, NULL);
+    mce_display_add_state_changed_handler(mce_display, display_cb, NULL);
+
+    mce_tklock_add_valid_changed_handler(mce_tklock, tklock_cb, NULL);
+    mce_tklock_add_mode_changed_handler(mce_tklock, tklock_cb, NULL);
+    mce_tklock_add_locked_changed_handler(mce_tklock, tklock_cb, NULL);
+
+    mce_inactivity_add_valid_changed_handler(mce_inactivity, inactivity_cb, NULL);
+    mce_inactivity_add_status_changed_handler(mce_inactivity, inactivity_cb, NULL);
+
+    if( argc > 1 )
+        timeout_s = strtol(argv[1], NULL, 0);
+    if( timeout_s > 0)
+        timeout_id = g_timeout_add(timeout_s * 1000, quit_cb,  &timeout_id);
+    sigterm_id = g_unix_signal_add(SIGTERM, quit_cb, &sigterm_id);
+    sigint_id  = g_unix_signal_add(SIGINT, quit_cb,  &sigint_id);
+
+    printf("mainloop\n");
+    exitcode = mainloop_run();
+    printf("cleanup\n");
+
+    g_clear_handle_id(&sigterm_id, g_source_remove);
+    g_clear_handle_id(&sigint_id, g_source_remove);
+    g_clear_handle_id(&timeout_id, g_source_remove);
```

Comment:
` g_clear_handle_id` requires glib 2.56, the library itself is compatible with 2.32, it doesn't make sense for the example to require a version of glib higher than the library itself.

Reply:
For once I checked if something even semi-sane would be available in glib.
Anyway, g_clear_handle_id() calls replaced with something g_source_remove()